runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBoxOverlay.java

```diff
@@ -122,25 +141,56 @@ public Dimension render(Graphics2D graphics)
 
 		for (final LayoutableRenderableEntity child : panelComponent.getChildren())
 		{
-			if (child instanceof InfoBoxComponent)
+			assert child instanceof InfoBoxComponent;
+			final InfoBoxComponent component = (InfoBoxComponent) child;
+
+			// Create intersection rectangle
+			final Rectangle intersectionRectangle = new Rectangle(component.getBounds());
+			intersectionRectangle.translate(getBounds().x, getBounds().y);
+
+			if (intersectionRectangle.contains(mouse))
 			{
-				final InfoBoxComponent component = (InfoBoxComponent) child;
+				final String tooltip = component.getTooltip();
+				if (!Strings.isNullOrEmpty(tooltip))
+				{
+					tooltipManager.add(new Tooltip(tooltip));
+				}
 
-				if (!Strings.isNullOrEmpty(component.getTooltip()))
+				if (!menuOpen)
 				{
-					// Create intersection rectangle
-					final Rectangle intersectionRectangle = new Rectangle(component.getBounds());
-					intersectionRectangle.translate(getBounds().x, getBounds().y);
-
-					if (intersectionRectangle.contains(mouse))
-					{
-						tooltipManager.add(new Tooltip(component.getTooltip()));
-					}
+					hoveredComponent = component;
 				}
+				break;
 			}
 		}
 
 		panelComponent.getChildren().clear();
 		return dimension;
 	}
+
+	@Override
+	public List<OverlayMenuEntry> getMenuEntries()
+	{
+		// we dynamically build the menu options based on which infobox is hovered
+		return hoveredComponent == null ? Collections.emptyList() : hoveredComponent.getInfoBox().getMenuEntries();
+	}
+
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked menuOptionClicked)
+	{
+		if (menuOptionClicked.getMenuAction() != MenuAction.RUNELITE_INFOBOX)
+		{
+			return;
+		}
+
+		InfoBox infoBox = hoveredComponent.getInfoBox();
+		OverlayMenuEntry overlayMenuEntry = infoBox.getMenuEntries().stream()
+			.filter(me -> me.getOption().equals(menuOptionClicked.getMenuOption()))
+			.findAny()
+			.orElse(null);
+		if (overlayMenuEntry != null)
+		{
+			eventBus.post(new InfoBoxMenuClicked(overlayMenuEntry, infoBox));
```

Comment:
why not just invoke a listener attached to the infobox?

Reply:
It would require having the eventbus in the infobox which we can't do easily.